src/components/Args.tsx

```diff
@@ -1,26 +1,21 @@
-import React, { Component } from "react";
-import PropTypes from "prop-types";
-
+import React from "react";
 import { ASTNode } from "../ast";
 import { DropTarget } from "./DropTarget";
-import { span } from "../types";
 
 export type ArgsProps = {
   field: string;
   children: ASTNode[];
 };
 
-export default class Args extends Component<ArgsProps> {
-  render() {
-    let { children } = this.props;
-    const elems = [];
-    elems.push(<DropTarget key={"drop-0"} field={this.props.field} />);
-    children.forEach((child, index) => {
-      elems.push(child.reactElement({ key: "node" + index }));
-      elems.push(
-        <DropTarget key={"drop-" + (index + 1)} field={this.props.field} />
-      );
-    });
-    return elems;
-  }
-}
+const Args = (props: ArgsProps) => {
+  const { field, children } = props;
```

Comment:
If you want to be more concise, you can put the object destructuring into the parameter declaration:

```ts
const Args = ({ field, children }: ArgsProps) => {
```

I'm still on the fence of whether its better to use object destructuring for the props or to always reference them through `props.whatever`. In smaller components, the former seems preferable because it's so concise. For larger components where there are a lot of non-prop variables (like stuff returned by hooks and whatnot), it seems like the latter is preferable because its easier to see at a glance whether a particular variable is passed in as a prop or is something defined in the component function. I wonder if some of the more opinionated linting rulesets have an opinion about this.